### PR TITLE
[DEV-1459] Initial definition of switch state events [skip ci]

### PR DESCRIPTION
**Pull request by kgreav**
# Description

RPCs and protos for supporting switch state updates to EWB

# Test Steps

Build should be testing enough...

# Checklist

If any of these are not applicable, strikethrough the line `~like this~`. **Do not delete it!**. Let the reviewer decide if you should have done it.

### Code
- [ ] I have performed a self review of my own code (including checking issues raised when creating the PR).
- [ ] I have added/updated unit tests for these changes, and if not I have explained why they are not necessary.
- [ ] I have commented my code in any hard-to-understand or hacky areas.
- [ ] I have handled all new warnings generated by the compiler or IDE.
- [ ] I have rebased onto the target branch (usually main).

### Documentation
- [ ] I have updated the changelog.
- [ ] I have updated any documentation required for these changes.

# Breaking Changes
- [ ] I have considered if this is a breaking change and will communicate it with other team members if so.

Breaks existing switch state interface which was not in use so NBD.

### CODE REVIEW

**Comment by ryanjcoleman** [comment]
Task linked: [DEV-1459 Define protocol](https://app.clickup.com/t/6929263/DEV-1459)

File: proto/zepben/protobuf/ns/data/change-events.proto [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+    /**
+     * The type of event that occurred.
+     */
+    oneof event {
+        SwitchStateEvent switch = 3;
+        AddCutEvent addCut = 4;
+        RemoveCutEvent removeCut = 5;
+        AddJumperEvent addJumper = 6;
+        RemoveJumperEvent removeJumper = 7;
```
Comment (kgreav) [inline proto/zepben/protobuf/ns/data/change-events.proto:42]: i reckon this is probably correct for cuts and jumpers, but i don't have enough information to confirm that yet. At the very least the events are correct, it'll just be a question of what data is captured in them which we won't find out until later.

File: proto/zepben/protobuf/ns/data/change-status.proto [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+import "google/protobuf/timestamp.proto";
+
+/**
+ * A response indicating all items in the batch were applied successfully.
+ */
+message BatchSuccessful {
+}
+
+/**
+ * A response indicating all items in the batch were applied successfully.
```
Comment (kgreav) [inline proto/zepben/protobuf/ns/data/change-status.proto:24]: Comment is wrong - assumedly idea here is that "if paused, tell client that we're paused, but they shouldn't retry"

Comment (charlta) [reply proto/zepben/protobuf/ns/data/change-status.proto]: Yes this is a copy/paste error and has been fixed.

File: proto/zepben/protobuf/ns/data/change-events.proto (hunk 1 above)
Comment (charlta) [reply proto/zepben/protobuf/ns/data/change-events.proto:42]: Yeah was only meant to be placeholders with proper messages to be done as a future pass when we have details on what is required

**Review by charlta — APPROVED** [review]

**Pull request merged**